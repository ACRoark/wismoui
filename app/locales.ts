export const DEFAULT_LOCALE = 'en';

const PROD_LOCALES = {
  en: 'English',
};

const TEST_LOCALES = {
  en: 'English',
  es: 'Spanish',
};

export const SUPPORTED_LOCALES = process.env.NODE_ENV === 'production' ? PROD_LOCALES : TEST_LOCALES;
