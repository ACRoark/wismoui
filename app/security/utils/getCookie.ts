const getCookie = (name: string): string =>
  document.cookie.split('; ').reduce((previousValue: string, currentValue: string): string => {
    const parts = currentValue.split('=');

    return parts[0] === name ? decodeURIComponent(parts[1]) : previousValue;
  }, '');

export default getCookie;
