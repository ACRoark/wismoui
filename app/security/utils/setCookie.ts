interface ICookieOptions {
  days: number;
  path: string;
}

const setCookie = (name: string, value: string, options: ICookieOptions): void => {
  const expires = new Date(Date.now() + options.days * 864e5).toUTCString();

  document.cookie = `${name}=${encodeURIComponent(value)}; expires=${expires}; path=${options.path}`;
};

export default setCookie;
