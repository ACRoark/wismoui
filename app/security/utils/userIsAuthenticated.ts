import useAuth from 'hooks/useAuth';

const userIsAuthenticated = (): boolean => {
  const { isAuthenticated } = useAuth();

  return isAuthenticated;
};

export default userIsAuthenticated;
