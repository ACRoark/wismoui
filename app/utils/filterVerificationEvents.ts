import IVerificationEvent from 'types/IVerificationEvent';
import filterVerificationEventStatusUpdatesByActive from './filterVerificationEventStatusUpdatesByActive';
import filterVerificationEventStatusUpdatesByVerificationCodes from './filterVerificationEventStatusUpdatesByVerificationCodes';

const filterVerificationEvents = (verificationEvents: IVerificationEvent[], verificationCodes: string[]): IVerificationEvent[] => {
  const filteredVerificationEvents: IVerificationEvent[] = [];
  verificationEvents.forEach((event: IVerificationEvent): void => {
    const activeVerificationEventStatusUpdates = filterVerificationEventStatusUpdatesByActive(event.verificationEventStatusUpdates);
    const verificationEventStatusUpdatesByVerificationCodes =
      filterVerificationEventStatusUpdatesByVerificationCodes(activeVerificationEventStatusUpdates, verificationCodes);

    if (verificationEventStatusUpdatesByVerificationCodes.length > 0) {
      const filteredEvent = {
        ...event,
        verificationEventStatusUpdates: verificationEventStatusUpdatesByVerificationCodes
      };

      filteredVerificationEvents.push(filteredEvent);
    }
  });

  return filteredVerificationEvents;
};

export default filterVerificationEvents;
