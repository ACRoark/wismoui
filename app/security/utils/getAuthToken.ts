import getCookie from './getCookie';

const getAuthToken = (): string => getCookie('token');

export default getAuthToken;
