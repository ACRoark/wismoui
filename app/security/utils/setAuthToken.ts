import setCookie from './setCookie';

const setAuthToken = (value?: string): void => {
  if (value) {
    setCookie('token', `Bearer ${value}`, { days: 1, path: '/' });
  }
};

export default setAuthToken;
