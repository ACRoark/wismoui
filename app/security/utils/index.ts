import getAuthToken from './getAuthToken';
import setAuthToken from './setAuthToken';
import userIsAuthenticated from './userIsAuthenticated';

export { getAuthToken, setAuthToken, userIsAuthenticated };
