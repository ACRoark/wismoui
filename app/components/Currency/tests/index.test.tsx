import React from 'react';

import { SUPPORTED_LOCALES } from 'locales';

import { createSnapshotWithIntl, renderWithIntl } from 'testing/utils';

import Currency from '..';

const currencyTypes = ['CAD', 'MXN', 'USD'];

const expectComponentToMatchSnapshot = (currencyType: string, value: string | number, language: string): void => {
  const tree = createSnapshotWithIntl(<Currency type={currencyType} value={value} />, language);

  expect(tree).toMatchSnapshot();
};

describe('<Currency />', (): void => {
  const languages = Object.keys(SUPPORTED_LOCALES);

  languages.forEach((language: string): void => {
    currencyTypes.forEach((currencyType: string): void => {
      it(`should not log errors in console when the currency type is ${currencyType} and the language is ${language}`, (): void => {
        const spy = jest.spyOn(global.console, 'error');

        renderWithIntl(<Currency type="USD" value={1000} />, language);

        expect(spy).not.toHaveBeenCalled();
      });

      it(`should render correctly when currency has value of number 1000 and the currency type is ${currencyType} (${language})`, (): void =>
        expectComponentToMatchSnapshot(currencyType, 1000, language));

      it(`should render correctly when currency has value of string ".59" and the currency type is ${currencyType} (${language})`, (): void =>
        expectComponentToMatchSnapshot(currencyType, '.59', language));

      it(`should render correctly when currency has value of string "1" and the currency type is ${currencyType} (${language})`, (): void =>
        expectComponentToMatchSnapshot(currencyType, '1', language));

      it(`should render correctly when currency has value of number -1.23 and the currency type is ${currencyType} (${language})`, (): void =>
        expectComponentToMatchSnapshot(currencyType, -1.23, language));
    });
  });
});
