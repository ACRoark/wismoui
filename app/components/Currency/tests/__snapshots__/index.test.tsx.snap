// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Currency /> should render correctly when currency has value of number -1.23 and the currency type is CAD (en) 1`] = `"-CA$1.23"`;

exports[`<Currency /> should render correctly when currency has value of number -1.23 and the currency type is CAD (es) 1`] = `"-1,23 CA$"`;

exports[`<Currency /> should render correctly when currency has value of number -1.23 and the currency type is MXN (en) 1`] = `"-MX$1.23"`;

exports[`<Currency /> should render correctly when currency has value of number -1.23 and the currency type is MXN (es) 1`] = `"-1,23 MXN"`;

exports[`<Currency /> should render correctly when currency has value of number -1.23 and the currency type is USD (en) 1`] = `"-$1.23"`;

exports[`<Currency /> should render correctly when currency has value of number -1.23 and the currency type is USD (es) 1`] = `"-1,23 US$"`;

exports[`<Currency /> should render correctly when currency has value of number 1000 and the currency type is CAD (en) 1`] = `"CA$1,000.00"`;

exports[`<Currency /> should render correctly when currency has value of number 1000 and the currency type is CAD (es) 1`] = `"1000,00 CA$"`;

exports[`<Currency /> should render correctly when currency has value of number 1000 and the currency type is MXN (en) 1`] = `"MX$1,000.00"`;

exports[`<Currency /> should render correctly when currency has value of number 1000 and the currency type is MXN (es) 1`] = `"1000,00 MXN"`;

exports[`<Currency /> should render correctly when currency has value of number 1000 and the currency type is USD (en) 1`] = `"$1,000.00"`;

exports[`<Currency /> should render correctly when currency has value of number 1000 and the currency type is USD (es) 1`] = `"1000,00 US$"`;

exports[`<Currency /> should render correctly when currency has value of string ".59" and the currency type is CAD (en) 1`] = `"CA$0.59"`;

exports[`<Currency /> should render correctly when currency has value of string ".59" and the currency type is CAD (es) 1`] = `"0,59 CA$"`;

exports[`<Currency /> should render correctly when currency has value of string ".59" and the currency type is MXN (en) 1`] = `"MX$0.59"`;

exports[`<Currency /> should render correctly when currency has value of string ".59" and the currency type is MXN (es) 1`] = `"0,59 MXN"`;

exports[`<Currency /> should render correctly when currency has value of string ".59" and the currency type is USD (en) 1`] = `"$0.59"`;

exports[`<Currency /> should render correctly when currency has value of string ".59" and the currency type is USD (es) 1`] = `"0,59 US$"`;

exports[`<Currency /> should render correctly when currency has value of string "1" and the currency type is CAD (en) 1`] = `"CA$1.00"`;

exports[`<Currency /> should render correctly when currency has value of string "1" and the currency type is CAD (es) 1`] = `"1,00 CA$"`;

exports[`<Currency /> should render correctly when currency has value of string "1" and the currency type is MXN (en) 1`] = `"MX$1.00"`;

exports[`<Currency /> should render correctly when currency has value of string "1" and the currency type is MXN (es) 1`] = `"1,00 MXN"`;

exports[`<Currency /> should render correctly when currency has value of string "1" and the currency type is USD (en) 1`] = `"$1.00"`;

exports[`<Currency /> should render correctly when currency has value of string "1" and the currency type is USD (es) 1`] = `"1,00 US$"`;
